File: baekjoon/js/1436.js
const fs = require("fs");
const buf = fs.readFileSync(0, "utf-8").toString().trim();

const n = Number(buf);

let count = 1;
let num = 666;

while (true) {
  if (count === n) {
    console.log(num);
    break;
  }

  num++;

  if (String(num).includes("666")) {
    count++;
  }
}

File: baekjoon/rust/1436.rs
use std::io::stdin;

fn main() {
    let mut buf = String::new();
    stdin().read_line(&mut buf).unwrap();

    let n: u32 = buf.trim().parse().unwrap();

    let mut num = 665;

    for _ in 0..n {
        num += 1;

        while !is_contains_666(num) {
            num += 1;
        }
    }

    print!("{}", num);
}

fn is_contains_666(mut num: u32) -> bool {
    while num > 665 {
        if (num - 666) % 1000 == 0 {
            return true;
        }

        num /= 10;
    }

    false
}
